x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/execution/search/Querier.java

```diff
@@ -629,11 +631,23 @@ protected void handleResponse(SearchResponse response, ActionListener<Page> list
 
         private HitExtractor createExtractor(FieldExtraction ref) {
             if (ref instanceof SearchHitFieldRef f) {
-                return new FieldHitExtractor(f.name(), f.getDataType(), cfg.zoneId(), f.hitName(), multiValueFieldLeniency);
+                return new FieldHitExtractor(
+                    f.name(),
+                    f.getDataType(),
+                    cfg.zoneId(),
+                    f.hitName(),
+                    multiValueFieldLeniency ? LENIENT : NONE
```

Comment:
Instead of having this ternary operator called twice, why not using MultiValueSupport instead of `multiValueFieldLeniency` in `SearchHitActionListener`? Meaning `private final MultiValueSupport multiValueFieldLeniency;` and in constructor `this.multiValueFieldLeniency = cfg.multiValueFieldLeniency() ? LENIENT : NONE;`